commit dbe8e38481d2b8ad44d9ca5a3e93a2e89c2bf317
Author: Scott Koranda <scott.koranda@ligo.org>
Date:   Mon Apr 22 14:00:15 2013 -0500

    Fix to DboSource.php in CakePHP release (CO-446)

diff --git a/lib/Cake/Model/Datasource/DboSource.php b/lib/Cake/Model/Datasource/DboSource.php
index 57368b4..6ffc2cd 100644
--- a/lib/Cake/Model/Datasource/DboSource.php
+++ b/lib/Cake/Model/Datasource/DboSource.php
@@ -1217,7 +1217,9 @@ class DboSource extends DataSource {
 								} else {
 									$db = ConnectionManager::getDataSource($deepModel->useDbConfig);
 								}
-								$db->queryAssociation($linkModel, $deepModel, $type1, $assoc1, $assocData1, $queryData, true, $fetch, $recursive - 1, $tmpStack);
+								if (method_exists($db, 'queryAssociation')){
+									$db->queryAssociation($linkModel, $deepModel, $type1, $assoc1, $assocData1, $queryData, true, $fetch, $recursive - 1, $tmpStack);
+								}
 							}
 						}
 					}
@@ -1291,7 +1293,9 @@ class DboSource extends DataSource {
 									} else {
 										$db = ConnectionManager::getDataSource($deepModel->useDbConfig);
 									}
-									$db->queryAssociation($linkModel, $deepModel, $type1, $assoc1, $assocData1, $queryData, true, $fetch, $recursive - 1, $tmpStack);
+									if (method_exists($db, 'queryAssociation')) {
+										$db->queryAssociation($linkModel, $deepModel, $type1, $assoc1, $assocData1, $queryData, true, $fetch, $recursive - 1, $tmpStack);
+									}
 								}
 							}
 						}
